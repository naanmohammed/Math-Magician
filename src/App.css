body {
  padding: 0 15px;
}

.navBar {
  font-size: 20px;
  font-weight: bold;
  display: flex;
  flex-wrap: wrap;
  gap: 10px;
  justify-content: space-between;
}

.navHeader {
  font-size: 48px;
  text-decoration: none;
  color: black;
}

.menu {
  display: flex;
  margin-top: 25px;
  gap: 10px;
}

.menuItem {
  color: blue;
}

.calculatorPage {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.container {
  margin-top: 35px;
  width: 70%;
}

.calculator-container {
  width: 100%;
}

.calculatorPage h2 {
  width: 50%;
  font-size: 30px;
}

.quotesPage h2 {
  padding-top: 10rem;
  font-size: 48px;
}
