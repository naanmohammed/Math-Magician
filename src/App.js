import { Outlet } from 'react-router';
import './App.css';
import Homepage from './pages/homepage';

export default function App() {
  return (
    <>
      <Homepage />
      <Outlet />
    </>
  );
}
