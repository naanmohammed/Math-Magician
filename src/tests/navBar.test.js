import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import NavBar from '../components/navBar';
import ChildComponent from '../components/childComponent';

describe('App', () => {
  it('renders the App component', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('ChildComponent', () => {
  const mockHandleButtonClick = jest.fn();
  const mockCalcData = {
    total: '10',
    next: null,
    operation: '+',
  };

  it('renders the ChildComponent', () => {
    const { container } = render(
      <ChildComponent calcData={mockCalcData} handleButtonClick={mockHandleButtonClick} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('NavBar', () => {
  it('renders the NavBar', () => {
    const { container } = render(<MemoryRouter><NavBar /></MemoryRouter>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
