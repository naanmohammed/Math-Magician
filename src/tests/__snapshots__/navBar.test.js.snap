// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App component 1`] = `
<div
  class="home"
>
  <div
    class="row"
  >
    <h1>
      Welcome to our page !
    </h1>
    <p>
       
      Lorem ipsum dolor sit amet. Eos nulla veritatis et quis voluptatem sed omnis eveniet! Et vero sint sit minima eligendi aut commodi sapiente est dolor quisquam. Rem molestiae quidem in labore eaque sed ipsum molestiae  rerum. Qui enim porro et quam veritatis qui temporibus possimus? Vel omnis velit eos ratione voluptas et quis provident sit voluptates sint aut velit magni! Ut asperiores omnis aut facilis soluta et cumque tenetur .
       
    </p>
    <p>
       
      Lorem ipsum dolor sit amet. Eos nulla veritatis et quis voluptatem sed omnis eveniet! Et vero sint sit minima eligendi aut commodi sapiente est dolor quisquam. Rem molestiae quidem in labore eaque sed ipsum molestiae  rerum. Qui enim porro et quam veritatis qui temporibus possimus? Vel omnis velit eos ratione voluptas et quis provident sit voluptates sint aut velit magni! Ut asperiores omnis aut facilis soluta et cumque tenetur .
       
    </p>
  </div>
</div>
`;

exports[`ChildComponent renders the ChildComponent 1`] = `
<div
  class="calculator-container"
>
  <input
    class="calculator-input-field"
    id="myInput"
    name="name"
    pattern="[0-9+\\\\-*/(). ]+"
    type="text"
    value="+"
  />
  <div
    class="calculator-buttons"
  >
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="AC"
        type="button"
      >
        AC
      </button>
      <button
        class="calculator-button  "
        id="+/-"
        type="button"
      >
        +/-
      </button>
      <button
        class="calculator-button  "
        id="%"
        type="button"
      >
        %
      </button>
      <button
        class="calculator-button math-button-orange "
        id="รท"
        type="button"
      >
        รท
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="7"
        type="button"
      >
        7
      </button>
      <button
        class="calculator-button  "
        id="8"
        type="button"
      >
        8
      </button>
      <button
        class="calculator-button  "
        id="9"
        type="button"
      >
        9
      </button>
      <button
        class="calculator-button math-button-orange "
        id="x"
        type="button"
      >
        x
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="4"
        type="button"
      >
        4
      </button>
      <button
        class="calculator-button  "
        id="5"
        type="button"
      >
        5
      </button>
      <button
        class="calculator-button  "
        id="6"
        type="button"
      >
        6
      </button>
      <button
        class="calculator-button math-button-orange "
        id="-"
        type="button"
      >
        -
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="1"
        type="button"
      >
        1
      </button>
      <button
        class="calculator-button  "
        id="2"
        type="button"
      >
        2
      </button>
      <button
        class="calculator-button  "
        id="3"
        type="button"
      >
        3
      </button>
      <button
        class="calculator-button math-button-orange "
        id="+"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  zero-button"
        id="0"
        type="button"
      >
        0
      </button>
      <button
        class="calculator-button  "
        id="."
        type="button"
      >
        .
      </button>
      <button
        class="calculator-button  "
        id="="
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`NavBar renders the NavBar 1`] = `
<div
  class="navBar"
>
  <a
    class="navHeader"
    href="/"
  >
    Math Magicians
  </a>
  <div
    class="menu"
  >
    <a
      class="menuItem"
      exact="true"
      href="/"
    >
      Home |
    </a>
    <a
      class="menuItem"
      exact="true"
      href="/calculator"
    >
      Calculator |
    </a>
    <a
      class="menuItem"
      exact="true"
      href="/quote"
    >
      Quote
    </a>
  </div>
</div>
`;
